src/core/Toast/Toaster.tsx

```diff
@@ -51,22 +38,45 @@ export default class Toaster {
     category: ToastCategory,
     settings?: ToastOptions,
   ) {
-    this.addToast({
-      ...settings,
-      content,
-      category,
-    });
+    const lastId = this.toasts[0]?.id ?? 0;
+    const currentId = lastId + 1;
+    this.toasts = [
+      {
+        ...settings,
+        content,
+        category,
+        onRemove: () => {
+          this.removeToast(currentId);
+          settings?.onRemove;
+        },
+        id: currentId,
+        isVisible: true,
+      },
+      ...this.toasts,
+    ];
+
+    this.updateView();
   }
 
-  public closeAll(): void {
-    this.closeAllHandler();
+  private removeToast(id: number) {
+    this.toasts = this.toasts.filter((toast) => toast.id !== id);
+    this.updateView();
   }
 
-  private _addHandler(addHandler: (settings: ToastSettings) => void) {
-    this.addToast = addHandler;
+  private updateView() {
+    ReactDOM.render(
```

Comment:
so this will create a whole new renderer separate from the main app? Sure this is the best way? 🤔 

I'm surprised this is how we were doing it before too 😮

Reply:
One key difference is that render() will be called multiple times now that it's not in the constructor. ~~Might be bad for performance~~ Should be fine 😄 https://twitter.com/dan_abramov/status/691301224541503488